mods/default/functions.lua

```diff
@@ -554,13 +554,16 @@ function default.can_interact_with_node(player, pos)
 	-- is player wielding the right key?
 	local item = player:get_wielded_item()
 	if item:get_name() == "default:key" then
-		local key_meta = minetest.parse_json(item:get_metadata())
+		local key_meta = item:get_meta()
 		local secret = meta:get_string("key_lock_secret")
-		if secret ~= key_meta.secret then
-			return false
+
+		if key_meta:get_string("secret") == "" then
+			key_meta:set_string("secret", minetest.parse_json(item:get_metadata()).secret)
 		end
 
-		return true
```

Comment:
Note to reviewers: This line was removed to allow owners to access the node at any time - no matter what item is being hold